src/ProgressOnderwijsUtils/Pocos/SmallBatchInsertImplementation.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using ExpressionToCodeLib;
+using Microsoft.Data.SqlClient;
+using ProgressOnderwijsUtils.Collections;
+using static ProgressOnderwijsUtils.SafeSql;
+
+namespace ProgressOnderwijsUtils
+{
+    public static class SmallBatchInsertImplementation
+    {
+        internal const int ThresholdForUsingSqlBulkCopy = 6;
+
+        public static IEnumerable<T>? TrySmallBatchInsertOptimization<T>(SqlConnection sqlConn, BulkInsertTarget bulkInsertTarget, IEnumerable<T> pocos, CommandTimeout timeout)
+            where T : IReadImplicitly
+        {
+            if ((bulkInsertTarget.Options | SqlBulkCopyOptions.KeepNulls) != (BulkInsertTarget.DefaultOptionsCorrespondingToInsertIntoBehavior | SqlBulkCopyOptions.KeepNulls)) {

```

Comment:
Die ``KeepNulls`` zitten toch al in DefaultOptionsCorrespondingToInsertIntoBehavior?

Reply:
ja, maar nu zijn ze niet meer nodig - mag aan, mag uit.

Reply:
Ik snap wel dat het werkt, maar ik vind het niet heel leesbaar wat deze regel doet.

Reply:
Heb je een beter voorstel?

Reply:
```suggestion
            if (((BulkInsertTarget.DefaultOptionsCorrespondingToInsertIntoBehavior ^ bulkInsertTarget.Options) & ~SqlBulkCopyOptions.KeepNulls) != default) {
```
is dit beter?

Reply:
misschien is een commentaarregel die duidelijkmaakt waarom deze escape er is wel voldoende

Reply:
```suggestion
            if ((bulkInsertTarget.Options | SqlBulkCopyOptions.KeepNulls) != (BulkInsertTarget.DefaultOptionsCorrespondingToInsertIntoBehavior | SqlBulkCopyOptions.KeepNulls)) {
                // alleen heel specifieke options gedragen zich hetzelfde als "insert into", dus als iemand afwijkende options heeft gekozen: abort.
```